awx/ui_next/src/components/Search/Search.jsx

```diff
@@ -106,10 +108,19 @@ class Search extends React.Component {
     e.preventDefault();
 
     const { searchKey, searchValue } = this.state;
-    const { onSearch } = this.props;
+    const { onSearch, qsConfig } = this.props;
+
+    const isNonStringField = key =>
+      qsConfig.integerFields.filter(field => field === key).length ||
+      qsConfig.dateFields.filter(field => field === key).length;
```

Comment:
Since we know the key, you can probably just set a variable here instead of defining the function:

```
const isNonStringField =  qsConfig.integerFields.filter(field => field === searchKey).length ||
      qsConfig.dateFields.filter(field => field === searchKey).length;
const actualSearchKey = isNonStringField ? searchKey : `${searchKey}__icontains`;
```